python/rapidsmpf/rapidsmpf/statistics.pyx

```diff
@@ -23,14 +23,14 @@ cdef extern from *:
     ) {
         return stats.get_stat(name).count();
     }
-    std::size_t cpp_get_statistic_value(
+    double cpp_get_statistic_value(
         rapidsmpf::Statistics const& stats, std::string const& name
     ) {
         return stats.get_stat(name).value();
     }
     """
-    size_t cpp_get_statistic_count(cpp_Statistics stats, string name) nogil
-    double cpp_get_statistic_value(cpp_Statistics stats, string name) nogil
+    size_t cpp_get_statistic_count(cpp_Statistics stats, string name) except +
+    double cpp_get_statistic_value(cpp_Statistics stats, string name) except +
```

Comment:
We do still need to drop the gil because accessing the statistics acquires a lock, so this can now deadlock due to lock ordering.